python/ribasim/ribasim/model.py

```diff
@@ -224,7 +235,11 @@ def _apply_crs_function(self, function_name: str, crs: str) -> None:
     def node_table(self) -> NodeTable:
         """Compute the full sorted NodeTable from all node types."""
         df_chunks = [node.node.df for node in self._nodes()]
-        df = pd.concat(df_chunks)
+        df = (
+            pd.concat(df_chunks)
+            if df_chunks
+            else pd.DataFrame(index=pd.Index([], name="node_id"))
```

Comment:
This is to handle empty models, without any nodes right? Did you need this for anything? I don't think we test empty models.

I think this fails, just like before, but now with:

```
>>> ribasim.model.NodeTable(pd.Index([], name="node_id"))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: BaseModel.__init__() takes 1 positional argument but 2 were given
>>> ribasim.model.NodeTable(df=pd.Index([], name="node_id"))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "D:\repo\ribasim\Ribasim\.pixi\envs\default\Lib\site-packages\pydantic\main.py", line 193, in __init__
    self.__pydantic_validator__.validate_python(data, self_instance=self)
pydantic_core._pydantic_core.ValidationError: 1 validation error for NodeTable
df
  Value error, Expected <class 'pandas.core.series.Series'> to have name 'node_id', found 'None' [type=value_error, input_value=Index([], dtype='object', name='node_id'), input_type=Index]
    For further information visit https://errors.pydantic.dev/2.8/v/value_error
```

Wouldn't we need all columns with the right dtypes? With this we get an object dtype.

So perhaps best to leave out this change unless needed.

Reply:
Your examples are not what's happening in the code, you need to pass the dataframe as the df argument to the nodetable.

That said, this might not be needed in this PR, it's for the edge case when you read an empty geopackage. And when you call nodetable on a new model, which previously failed.

Reply:
Oh sorry I pasted two prompts, the first once was missing the `df` arguments, but the second one is not.

Reply:
Your second one passes an index, not a dataframe 😉

Reply:
Ha my bad. Let's merge!